tokio/src/runtime/task/core.rs

```diff
@@ -394,13 +395,13 @@ impl Header {
     }
 
     // safety: The caller must guarantee exclusive access to this field, and
-    // must ensure that the id is either 0 or the id of the OwnedTasks
+    // must ensure that the id is either `None` or the id of the OwnedTasks
     // containing this task.
-    pub(super) unsafe fn set_owner_id(&self, owner: u64) {
+    pub(super) unsafe fn set_owner_id(&self, owner: Option<NonZeroU64>) {
         self.owner_id.with_mut(|ptr| *ptr = owner);
     }
```

Comment:
This is never called with `None`, so you might as well just take a `NonZeroU64` as argument?

Reply:
Done.